rollout/trafficrouting/istio/istio.go

```diff
@@ -133,6 +133,10 @@ func (r *Reconciler) reconcileVirtualService(obj *unstructured.Unstructured, des
 		return nil, false, err
 	}
 
+	if err := validateHTTPRoutes(r.rollout, httpRoutes); err != nil {
```

Comment:
I feel like this should be happening when validating the entire rollout, not just when we are reconciling virtual services. Note that we will eventually have an admission web hook that needs to validate a rollout, so there needs to be a helper to statically lint a rollout.

Reply:
You may disregard this comment. Performing validation on referenced resources can/should happen during reconciliation time, and not via validating admission webhook.